aiohttp/web.py

```diff
@@ -221,6 +221,22 @@ def middlewares(self):
         return self._middlewares
 
     def make_handler(self, **kwargs):
+        try:
+            debug = kwargs['debug']
+        except KeyError:
+            pass
+        else:
+            if debug != self.debug:
+                raise ValueError(
+                    "The value of `debug` parameter conflicts with the debug "
+                    "settings of the `Application` instance. "
+                    "The application's debug mode setting should be used "
+                    "instead as a single point to setup a debug mode. "
+                    "For more information please check "
+                    "http://aiohttp.readthedocs.io/en/stable/"
+                    "web_reference.html#aiohttp.web.Application"
+                )
+        kwargs['debug'] = self.debug
```

Comment:
just call `self._handler_factory(self, self.router, debug=debug, loop=self.loop, **kwargs)` instead of adding the flag to `kwargs` back.